evm/utils.js

```diff
@@ -852,6 +861,73 @@ const mainProcessor = async (options, processCommand, save = true, catchErr = fa
         }
     }
 
+    if (options.parallel && chains.length > 1) {
```

Comment:
Please avoid making functions so complex. Use clear and concise helper functions to organize the code